src/components/App/index.tsx

```diff
@@ -1,19 +1,35 @@
-import React, { useState } from 'react';
+import React, { useEffect, useReducer } from 'react';
 import i18next from 'i18next';
 
+import { Context, reducer, INITIAL_STATE, actionCreators } from 'contexts/UserContext';
 import Routes from 'routes';
 import I18n from 'components/I18n';
+import LocalStorageService from 'services/LocalStorageService';
 
 import styles from './styles.module.scss';
 
 function App() {
-  const [language, setLanguage] = useState(i18next.language);
+  const [state, dispatch] = useReducer(reducer, INITIAL_STATE);
+  useEffect(() => {
+    const lang: string = LocalStorageService.getValue('lang');
+    if (lang) {
+      dispatch(actionCreators.setLanguage(lang));
+      i18next.changeLanguage(lang);
+    }
+    if (LocalStorageService.getValue('token')) {
+      dispatch(actionCreators.login(true));
+    } else {
+      dispatch(actionCreators.logout());
+    }
+  }, [state.language, dispatch]);
```

Comment:
I think you should divide this useEffect in two, since when a language change is made, the login should not be verified again.